File: server/request_header.py
from response_header import ResponseHeader

class RequestHeader(ResponseHeader):
  def __init__(self, client_id, version, code, payload_size):
    super().__init__(version, code, payload_size)
    self.client_id = client_id

  def get_client_id(self):
    return self.client_id

File: client/request_header.cpp
#include <cstring>
#include "request_header.h"

RequestHeader::RequestHeader(std::string client_id, uint8_t version, uint16_t code, uint32_t payload_size) 
  : ResponseHeader(version, code, payload_size) {
  strncpy(this->client_id, client_id.c_str(), 16);
}
